src/ai/backend/manager/scheduler/dispatcher.py

```diff
@@ -471,6 +472,38 @@ async def _check_predicates() -> List[Tuple[str, Union[Exception, PredicateResul
                     )
                     has_failure = True
 
+            async def _check_predicates_hook() -> HookResult:
+                async with self.db.begin_session() as db_sess:
```

Comment:
Could we make this a read-only session if it does not violate any intended use case?